Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/shape_smoothing.h

```diff
@@ -0,0 +1,408 @@
+
+  std::size_t nb_iterations = choose_param(get_param(np, internal_np::number_of_iterations), 1);
+
+  #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+  t.stop();
+  std::cout << "\rSmoothing parameters done ("<< t.time() <<" sec)" << std::endl;
+  std::cout << "Remesher construction...";
+  std::cout.flush();
+  t.reset(); t.start();
+  #endif
+
+  internal::Curvature_flow<PolygonMesh, VertexPointMap, VCMap, GeomTraits>
+          curvature_remesher(pmesh, vpmap, vcmap);
+
+  #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+  t.stop();
+  std::cout << " done ("<< t.time() <<" sec)." << std::endl;
+  std::cout << "Removing degenerate faces..." << std::endl;
+  t.reset(); t.start();
+  #endif
+
+  curvature_remesher.remove_degenerate_faces();
+
+  #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+  t.stop();
+  std::cout << " done ("<< t.time() <<" sec)." << std::endl;
+  std::cout << "Initializing..." << std::endl;
+  t.reset(); t.start();
+  #endif
+
+  curvature_remesher.init_smoothing(faces);
+
+  #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+  t.stop();
+  std::cout << " done ("<< t.time() <<" sec)." << std::endl;
+  std::cout << "#iter = " << nb_iterations << std::endl;
+  std::cout << "Shape smoothing..." << std::endl;
+  t.reset(); t.start();
+  #endif
+
+  for(std::size_t i=0; i<nb_iterations; ++i)
+  {
+
+    #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+        std::cout << " * Iteration " << (i + 1) << " *" << std::endl;
+    #endif
+
+    curvature_remesher.curvature_smoothing();
+
+  }
+
+  #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+  t.stop();
+  std::cout << "Shape smoothing done in ";
+  std::cout << t.time() << " sec." << std::endl;
+  std::cout<<std::endl;
+  #endif
+}
+
+/*!
+* \ingroup PMP_meshing_grp
+* smooths the overall shape of the mesh by using the mean curvature flow.
+* The effect depends on the curvature of each area and on a time step which
+* represents the amount by which vertices are allowed to move.
+* The result is a conformal map of the initial surface and converges to a sphere.
+*
+*
+* @tparam PolygonMesh model of `MutableFaceGraph`.
+*         If `PolygonMesh` has an internal property map for `CGAL::face_index_t`,
+*         and no `face_index_map` is given
+*         as a named parameter, then the internal one should be initialized.
+* @tparam FaceRange range of `boost::graph_traits<PolygonMesh>::%face_descriptor`,
+*         model of `Range`. Its iterator type is `ForwardIterator`.
+* @tparam NamedParameters a sequence of \ref pmp_namedparameters "Named Parameters"
+*
+* @param pmesh a polygon mesh with triangulated surface patches to be smoothed.
+* @param faces the range of triangular faces defining one or several surface patches to be smoothed.
+* @param time a time step that corresponds to the speed by which the surface is smoothed.
+*        A larger time step results in faster convergence but details may be distorted to a greater effect
+*        compared to more iterations with a smaller step. Typical values scale at the region (1e-6, 1).
```

Comment:
"at the region" should be "in the interval", imho (and if `1` if included, the interval should be `[1e-6, 1]`)